pkg/daemon/daemon.go

```diff
@@ -1325,44 +1385,83 @@ func (dn *Daemon) triggerUpdateWithMachineConfig(currentConfig, desiredConfig *m
 	return dn.update(currentConfig, desiredConfig)
 }
 
+// maybeValidateOnDiskState will conditionally run validateOnDiskState
+// depending on the presence of the forcefile and provides an indication that
+// validation was skipped. If the forcefile is present, this will *not* run
+// validateOnDiskState.
+func (dn *Daemon) maybeValidateOnDiskState(expectedConfig *mcfgv1.MachineConfig) (bool, error) {
```

Comment:
Can this be consolidated into validateOnDiskState or elsewhere? This func seems confusing - the bool is false if you do validateOnDiskState? Seems counterintuitive but also really not clear what's happening would prefer to just have this if check condensed as Jerry requested elsewhere.